scripts/train/train.py

```diff
@@ -164,170 +165,315 @@ def build_dataloader(cfg: DictConfig, tokenizer: PreTrainedTokenizerBase,
+                                                       'console_log_interval',
+                                                       must_exist=False,
+                                                       default_value='1ba')
+    device_train_microbatch_size: Union[str, int] = pop_config(
+        cfg,
+        'device_train_microbatch_size',
+        must_exist=False,
+        default_value='auto')
+    eval_subset_num_batches: int = pop_config(cfg,
+                                              'eval_subset_num_batches',
+                                              must_exist=False,
+                                              default_value=-1)
+    eval_first: bool = pop_config(cfg,
+                                  'eval_first',
+                                  must_exist=False,
+                                  default_value=False)
+    load_path: str = pop_config(cfg,
+                                'load_path',
+                                must_exist=False,
+                                default_value=None)
+    load_weights_only: bool = pop_config(cfg,
+                                         'load_weights_only',
+                                         must_exist=False,
+                                         default_value=False)
+    load_ignore_keys: Optional[List[str]] = pop_config(cfg,
+                                                       'load_ignore_keys',
+                                                       must_exist=False,
+                                                       default_value=None)
+    # Enable autoresume from model checkpoints if possible
+    autoresume_default = not save_overwrite and not save_weights_only
+    autoresume = pop_config(cfg,
+                            'autoresume',
+                            must_exist=False,
+                            default_value=autoresume_default)
+
+    # Pop known unused parameters.
+    pop_config(cfg, 'data_local', must_exist=False)
+    pop_config(cfg, 'data_remote', must_exist=False)
+    pop_config(cfg, 'global_seed', must_exist=False)
+    pop_config(cfg, 'global_train_batch_size', must_exist=False)
+    pop_config(cfg, 'n_gpus', must_exist=False)
+    pop_config(cfg, 'device_train_grad_accum', must_exist=False)
+
+    # Warn users for unused parameters
+    for key in cfg:
+        warnings.warn(
+            f'Unused parameter {key} found in cfg. Please check your yaml to ensure this parameter is necessary.'
+        )
+
+    # Warn if fsdp is enabled but user only has 1 GPU
     if dist.get_world_size() == 1 and fsdp_config is not None:
         warnings.warn(
             'FSDP is not applicable for single-GPU training. Reverting to DDP.')
-        cfg.pop('fsdp_config')
         fsdp_config = None
 
-    init_context = process_init_device(cfg.model, fsdp_config)
+    # Initialize context
+    init_context = process_init_device(model_config, fsdp_config)
+    if fsdp_config:
```

Comment:
same, we can remove this and just log once after trainer init

Reply:
done